### ISSUE TITLE
Kafka SSL smoke test is flaky

### ISSUE THREAD

**Issue by guswynn** [MaterializeInc/materialize#15173]
### Buildkite link

https://buildkite.com/materialize/tests/builds/42569#0183a43e-e5d4-4b43-8148-476955ab0669 (and others)

### Relevant log output

```source
Received error from Kafka stream consumer: deadline has elapsed
multi.td:69:1: error: missing record 0: Record {
    headers: [],
    key: None,
    value: Some(
        {
            (
                "before",
                Union {
                    index: 0,
                    inner: Null,
                    n_variants: 2,
                    null_variant: Some(
                        0,
                    ),
                },
            ),
            (
                "after",
                Union {
                    index: 1,
                    inner: {
                        (
                            "a",
                            Long(1),
                        ),
                    },
                    n_variants: 2,
                    null_variant: Some(
                        0,
                    ),
                },
            ),
        },
    ),
}
     |
  68 |
  69 | $ kafka-verify format=avro sink=materialize.public.snk sort-messages=true
     | ^
smoketest.td
```

### Additional thoughts

unclear when it started happening

**Comment by cjubb39** [comment]
Likely #15146.  Will investigate

**Comment by cjubb39** [comment]
My best guess at the moment is that we're incorrectly picking an AS OF for the sink now that the order of creating the storaged process and updating the catalog is inverted.  I'm not sure why this is only showing up in the SSL smoke test (maybe also the SASL smoketest for which I've seen some errors).  I don't have much evidence to back this up (other than the logs show the sink is progressing as usual but I don't see a message about writing out any transactions)

I don't have a huge amount of confidence I'll find this in the next half hour so I'm going to revert #15146 and run this test a bit more thoroughly before re-merging.

**Comment by guswynn** [comment]
@cjubb39 I might have also seen some flakes in the sasl smoketest

If its not flaky tomorrow after the revert we will know!

**Comment by cjubb39** [comment]
It stopped flaking after revert.  It was re-merged in #15235 after I fixed another issue.  And does not seem to be flaky at the moment
